### DISCUSSION TITLE
strimzi kafka operator crd and code structure

### DISCUSSION THREAD

**Discussion by lanzhiwang** [strimzi/strimzi-kafka-operator#4173] [General]
The strimzi kafka operator is written in java language, which is different from the operators defined by common operator-sdk. Can you introduce how crd is defined and registered, or if I want to add a crd, what should I do?

**Comment by scholzj** [comment]
First of all, before adding any new CRDs, you should maybe explain what are you trying to do to make sure you don't waste your effort. For example by writing a proposal: https://github.com/strimzi/proposals

The Strimzi CRDs are defined in the `api` module as a plain Java classes. If you look there for example into the `Kafka` class, you should see how it is defined and how it uses the different other classes for the fields etc. Once the CRD is defined there in the Java classes, the `crd-generator` module takes it and generates from it the CRD YAML files and Docs.

Does this help?
